bosh-director/lib/bosh/director/vm_deleter.rb

```diff
@@ -17,7 +18,11 @@ def delete_for_instance(instance)
 
     def delete_vm(vm_cid)
       @logger.info('Deleting VM')
-      @error_ignorer.with_force_check { @cloud.delete_vm(vm_cid) }
+      @error_ignorer.with_force_check {
+        unless @enable_virtual_delete_vm
+          @cloud.delete_vm(vm_cid)
```

Comment:
The default value of @enable_virtual_delete_vm is false, so if you did not set @enable_virtual_delete_vm = true explicitly, bosh director will invoke @cloud.delete_vm to delete vm from infrastructure, otherwise, bosh will keep it.